code/modules/mod/modules/modules_engineering.dm

```diff
@@ -260,3 +260,75 @@
 #undef EXTINGUISHER
 #undef NANOFROST
 #undef METAL_FOAM
+
+/obj/item/mod/module/drone
+	name = "MOD Personal Artificial Intelligence Storage"
+	desc = "A module that can be used to store a personal AI device in. \
+			Will convert the personal AI into a construction drone to help \
+			the owner with any repair work"
+	icon_state = "tether"
+	module_type = MODULE_USABLE
+	complexity = 1
+	use_power_cost = DEFAULT_CHARGE_DRAIN
+	incompatible_modules = list(/obj/item/mod/module/drone)
+	cooldown_time = 3 SECONDS
+	allow_flags = MODULE_ALLOW_INACTIVE
+	/// The drone that we currently have stored
+	var/mob/living/silicon/robot/drone/stored_drone
+	/// The drone that is linked to the mod
+	var/mob/living/silicon/robot/drone/linked_drone
+	/// The mob that summoned the drone. Used to clear the trait
+	var/mob/living/carbon/human/summoner
+
+/obj/item/mod/module/drone/Destroy()
+	clear_references()
+	linked_drone.pathfind_to_dronefab()
+	return ..()
+
+/obj/item/mod/module/drone/on_use()
+	. = ..()
+	if(stored_drone)
+		stored_drone.forceMove(get_turf(mod.wearer))
+	if(!linked_drone)
+		poll_ghosts_for_drone(mod.wearer)
+
+/obj/item/mod/module/drone/proc/create_new_drone(mob/living/user, mob/to_be_droned)
+	if(stored_drone)
+		to_chat(user, "<span class='notice'>There is already a drone stored in [src]!</span>")
+		return FALSE
+	if(!to_be_droned.mind)
+		to_chat(user, "<span class='notice'>[to_be_droned] does not have an inhabited AI!</span>")
+		return FALSE
+	if(HAS_MIND_TRAIT(mod.wearer, TRAIT_CREATED_DRONE))
```

Comment:
make sure mod.wearer isnt null before this, they could take off the suit in the 10 seconds of polling

Reply:
Will do